01-intro.Rmd

```diff
@@ -0,0 +1,127 @@
+
+
+## Workflows {-}
+
+### GitHub Workflow {-}
+
+```{r fig.align='center', echo=FALSE, include=identical(knitr:::pandoc_to(), 'html')}
+knitr::include_graphics('images/githubflow.png', dpi = NA)
+```
+&nbsp;
+
+This is our recommended workflow. The workflow is branch-based and involves keeping the master branch clean and 'production ready'/fully-functional at all times. 
+
+1. **Create a branch** - topic/per-person branches created from the master allow teams to contribute to many parallel efforts. Short-lived topic branches, in particular, keep teams focussed and results in quick ships.
+2. **Add commits** - these are snapshots of development efforts within a branch and create safe, revertible points in the project's history.
+3. **Open a pull request** - pull requests publicise the project's ongoing efforts and set the tone for a transparent development process.
+4. **Discuss and review code** - teams participate in code reviews by commenting, testing, and reviewing open pull requests. Code review is at the centre of an open and participatory culture. Pull requests should be approved by at least one collaborator - we recommend you assign a specific reviewer(s) to check the work. In order to ensure the master branch remains 'production ready' we also recommend that the master branch is protected and any changes are tested by the reviewer before approving.
+5. **Merge** - upon clicking merge, GitHub automatically performs the equivalent of a local `git merge` operation. GitHub also keeps the entire branch development history on the merged pull request.
+6. **Deploy** - teams can choose the best release cycles or incorporate continuous integration tools and operate with the assurance that code on the deployment branch has gone through a robust workflow.
+
+
+## Setup {-}
+
+### Git {-}
+
+*These intructions should only need to be followed once or when setting up a new device.*
+
+1. **Install Git** - Git is a free, open-source, software available from https://git-scm.com/. If you work within NHS National Services Scotland, request the software from IT through the [Service Portal](https://nhsnss.service-now.com/sp). Once authorised, this will allow you to download it from the Software Center on your machine. 
+
+This will install three applications on your computer, go to Start > Git (folder):
+
+* **Git Bash** - this is the command line (Shell) interface for git. This will allow you to enter git commands, i.e. `git init`, and can be launched by right clicking in a directory/folder and selecting "Git Bash Here".
+* **Git CMD** - this is a deprecated version, similar to the Bash interface, and shouldn't be used here.
+* **Git GUI** - this is git's version of a Graphical User Interface. This will perform the same functions as "Git Bash" by the user selecting options rather than having to type in the commands. This guide will outline using integrations on software like RStudio to utilise GUIs rather than this option.
```

Comment:
```suggestion
* **Git GUI** - this is git's version of a Graphical User Interface. This will perform the same functions as "Git Bash", but employs a point-and-click interface instead. This guide does not cover Git GUI and instead focuses on the integrated Git GUI in RStudio.
```